### DISCUSSION TITLE
Normalise projective coordinate

### DISCUSSION THREAD

**Discussion by pke** [paulmillr/noble-curves#125] [Q&A] [answered]
Java Bouncycastle has a function to normalise a projective coord.

> Normalization ensures that any projective coordinate is 1, and therefore that the x, y coordinates reflect those of the equivalent point in an affine coordinate system.
Returns:
a new ECPoint instance representing the same point, but with normalized coordinates

Is there a way to do that with this lib too?

I need to convert this Java code to JS:
`val sharedSecretKBytes: ByteArray = Bytes.bigIntToByteArray(sharedSecretK.normalize().xCoord.toBigInteger())`

**Comment by paulmillr** [comment]
Point.fromAffine(point.toAffine())

**Reply by pke** [reply]
Thanks for you quick response, @paulmillr 
so that would be `Point.fromAffine(sharedSecret.toAffine())` then?

**Reply by paulmillr** [reply, marked as answer]
If you have byte array (or hex) in `sharedSecret`, simply doing `Point.fromHex()` would create normalized `z=1` point.

If you have actual Point instance, then yes, your code.

**Reply by pke** [reply]
**edit:** Nevermind, my `toHex()` function that converted the `x` to a hex string, was faulty. It was not correctly padding the first byte with `0` with odd `x.toString(16)` results.

I seem to have 10% errors in that calculation here using a `brainpoolp256r1` curve.

```
const piccPk2 = Buffer.from(...)
const piccPk2ECPoint = curve.ProjectivePoint.fromHex(piccPk2.toString("hex"))
const sharedSecretK = piccPk2ECPoint.multiply(pcdSkX2)
const x = sharedSecretK.toAffine().x
```

The result in `x` I am getting for the same piccPk2 buffer and pcdSkX2 bigint is different from the Java results. I guess the `toAffine()` or `fromHex` are wrong?
